### PR TITLE
Specify replicates per sample when propagating

### PR DESCRIPTION
**Pull request by djcooke**

### CODE REVIEW

File: miso-web/src/main/webapp/scripts/hot_sample.js [hunk 1]
```diff
@@ -312,22 +312,23 @@ HotTarget.sample = (function() {
...
+              if (!value || value == "Delete external name, select a project, then re-enter external name." || value == '(...searching...)'
+                  || /First Receipt \(.*\)/.test(value) || !flat['externalName']) {
                 setOptions({
                   'renderer': Handsontable.renderers.TextRenderer
                 });
                 return;
-              };
-              
+              }
+              ;
```
Comment (sqwci) [inline miso-web/src/main/webapp/scripts/hot_sample.js]: MINOR Remove this empty statement. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1116)

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/BulkPropagateTableBackend.java [hunk 2]
```diff
@@ -1,20 +1,23 @@
...
+import uk.ac.bbsrc.tgac.miso.core.data.Identifiable;
+
 /**
  * Create a Handsontable for propagating a particular entity type
  *
  * @param <ParentModel> The database model for the entity from which the model can be propagated
  * @param <Dto> The DTO for the entity being propagated
  */
-public abstract class BulkPropagateTableBackend<ParentModel, Dto> extends BulkTableBackend<Dto> {
+public abstract class BulkPropagateTableBackend<ParentModel extends Identifiable, Dto> extends BulkTableBackend<Dto> {
```
Comment (sqwci) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/BulkPropagateTableBackend.java]: MINOR Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00119)
MINOR Rename this generic name to match the regular expression '^[A-Z][0-9]?$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00119)

**Comment by sqwci** [comment]
SonarQube analysis reported 25 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![CRITICAL][CRITICAL] 9 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 14 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [EditLibraryController.java#L549](https://github.com/TGAC/miso-lims/blob/4e056a720226f390487915ce48f066936e0d5648/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java#L549): Define a constant instead of duplicating this literal "title" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [EditLibraryController.java#L575](https://github.com/TGAC/miso-lims/blob/4e056a720226f390487915ce48f066936e0d5648/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java#L575): Define a constant instead of duplicating this literal "library" 4 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [EditLibraryController.java#L628](https://github.com/TGAC/miso-lims/blob/4e056a720226f390487915ce48f066936e0d5648/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java#L628): Define a constant instead of duplicating this literal "Libraries" 4 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [EditLibraryController.java#L854](https://github.com/TGAC/miso-lims/blob/4e056a720226f390487915ce48f066936e0d5648/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java#L854): Define a constant instead of duplicating this literal "Dilutions" 4 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [EditLibraryController.java#L902](https://github.com/TGAC/miso-lims/blob/4e056a720226f390487915ce48f066936e0d5648/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java#L902): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)
1. ![CRITICAL][CRITICAL] [EditLibraryController.java#L970](https://github.com/TGAC/miso-lims/blob/4e056a720226f390487915ce48f066936e0d5648/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java#L970): Define a constant instead of duplicating this literal "_POOL" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [EditSampleController.java#L694](https://github.com/TGAC/miso-lims/blob/4e056a720226f390487915ce48f066936e0d5648/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java#L694): Define a constant instead of duplicating this literal "sample" 4 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [EditSampleController.java#L894](https://github.com/TGAC/miso-lims/blob/4e056a720226f390487915ce48f066936e0d5648/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java#L894): Define a constant instead of duplicating this literal "Samples" 4 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![CRITICAL][CRITICAL] [EditSampleController.java#L943](https://github.com/TGAC/miso-lims/blob/4e056a720226f390487915ce48f066936e0d5648/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditSampleController.java#L943): Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [EditLibraryController.java#L462](https://github.com/TGAC/miso-lims/blob/4e056a720226f390487915ce48f066936e0d5648/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java#L462): Format specifiers should be used instead of string concatenation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3457)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/BulkPropagateTableBackend.java [hunk 3]
```diff
@@ -57,4 +60,34 @@ public final ModelAndView propagate(String idString, int replicates, ModelMap mo
     return prepare(model, true, "Create " + name + " from " + parentName, dtos);
   }
 
+  /**
+   * Create a view to propagate parents to new entities.
+   * 
+   * @param idString a comma-delimited list of parent IDs
+   * @param replicates the number of copies of each target that should be provided for parent
```
Comment (Kashif-S) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/BulkPropagateTableBackend.java]: This should be `replicatesString`.

**Pull request merged**